File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
      unordered_map<int,int>mp;
      int cnt = 0,elem = 0,prefix = 0;
      mp[0] = 1;
      for(int i=0;i<nums.size();i++){
          prefix += nums[i];
          elem = prefix - k;
          if(mp.find(elem) != mp.end()){
            cnt+=mp[elem];
          }
          mp[prefix]++;
      }  
      return cnt;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer>mp = new HashMap<>();
       int sum =0, cnt =0, r = 0 , l = 0;
  
       while(r<nums.length){
         
          sum += nums[r];
          int find = sum - k;
          if(sum == k) cnt++;
          if(mp.containsKey(find)){
             cnt += mp.get(find);
          }
           mp.put(sum, mp.getOrDefault(sum, 0) + 1);
          r++;
       }
       return cnt;
    }
}